server/kolide/options.go

```
@@ -0,0 +1,166 @@
+	ID uint
+	// Name the name of the option which must be unique
+	Name string
+	// Type of value expected for the option, db only
+	Type OptionType
+	// RawValue is string representation of option value, may be nil to
+	// indicate the option is not set
+	Value interface{} `db:"value"`
+	// ReadOnly if true, this option is required for Kolide to function
+	// properly and cannot be modified by the end user
+	ReadOnly bool `db:"read_only"`
+}
+
+// ValueAsString is a convenience method that converts the Value field of an
+// option to a string.
+func (opt *Option) ValueAsString() (val string, isDefined bool) {
+	if opt.Value == nil {
+		return "", false
+	}
+	switch opt.Value.(type) {
+	case []uint8:
+		return string(opt.Value.([]uint8)), true
+	case *string:
+		return *(opt.Value.(*string)), true
+	case string:
+		return opt.Value.(string), true
+	}
+	panic("unknown option type")
+}
+
+// OptionRequest contains options that are passed to modify options requests.
+type OptionRequest struct {
+	Options []Option `json:"options"`
+}
+
+type transform struct {
+	ID       uint        `json:"id"`
+	Name     string      `json:"name"`
+	Type     string      `json:"type"`
+	Value    interface{} `json:"value"`
+	ReadOnly bool        `json:"read_only"`
+}
+
+func (opt *Option) MarshalJSON() ([]byte, error) {
+	var val interface{}
+	if opt.Value == nil {
+		val = nil
+	} else {
+		strPtr, ok := opt.Value.(*string)
+		if !ok {
+			return nil, fmt.Errorf("option value is not expected type")
+		}
+		if strPtr == nil {
+			val = strPtr
+		} else {
+			switch opt.Type {
+			case OptionTypeString:
+				val = *strPtr
+			case OptionTypeInt:
+				i, err := strconv.ParseInt(*strPtr, 10, 64)
+				if err != nil {
+					return nil, err
+				}
+				val = i
+			case OptionTypeFlag:
+				b, err := strconv.ParseBool(*strPtr)
+				if err != nil {
+					return nil, err
+				}
+				val = b
+			default:
+				panic("unimplemented option type")
```

Why panic when we can return an error instead? Is the intention of this PR to implement all the option types? 

Also, the panic/error should contain the opt.Type it's reporting on. add a `: " + opt.Type` there.